vector/src/main/java/im/vector/riotx/core/platform/EllipsizingTextView.kt

```diff
@@ -1,421 +0,0 @@
-/*
- * Copyright (C) 2011 Micah Hainline
- * Copyright (C) 2012 Triposo
- * Copyright (C) 2013 Paul Imhoff
- * Copyright (C) 2014 Shahin Yousefi
- * Copyright 2020 New Vector Ltd
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- *
- */
-package im.vector.riotx.core.platform
-
-import android.content.Context
-import android.graphics.Canvas
-import android.graphics.Color
-import android.text.Layout
-import android.text.Spannable
-import android.text.SpannableString
-import android.text.SpannableStringBuilder
-import android.text.Spanned
-import android.text.StaticLayout
-import android.text.TextUtils.TruncateAt
-import android.text.TextUtils.concat
-import android.text.TextUtils.copySpansFrom
-import android.text.TextUtils.indexOf
-import android.text.TextUtils.lastIndexOf
-import android.text.TextUtils.substring
-import android.text.style.ForegroundColorSpan
-import android.util.AttributeSet
-import androidx.appcompat.widget.AppCompatTextView
-import androidx.core.content.withStyledAttributes
-import timber.log.Timber
-import java.util.ArrayList
-import java.util.regex.Pattern
-
-/*
- * Imported from https://gist.github.com/hateum/d2095575b441007d62b8
- *
- * Use it in your layout to avoid this issue: https://issuetracker.google.com/issues/121092510
- */
-
-/**
- * A [android.widget.TextView] that ellipsizes more intelligently.
- * This class supports ellipsizing multiline text through setting `android:ellipsize`
- * and `android:maxLines`.
- *
- *
- * Note: [TruncateAt.MARQUEE] ellipsizing type is not supported.
- * This as to be used to get rid of the StaticLayout issue with maxLines and ellipsize causing some performance issues.
- */
-class EllipsizingTextView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = android.R.attr.textViewStyle)
```

Comment:
It has been introduced to fix ellipsis, so is it ok to remove it now?

Reply:
It adds more issue than it solves